### ISSUE TITLE
collect CPU, memory and disk limits from cgroups and provide alarms using thresholds as percentages

### ISSUE THREAD

**Issue by sn-x** [netdata/netdata#2401]
In file src/sys_fs_cgroup.c, lines  2089, 2120

We are currently displaying all available cores on the host.
Would it be more appropriate if we only count the cores assigned to the guest?

**Comment by ktsaou** [comment]
I guess you mean this part: https://github.com/ktsaou/netdata/blob/fbd2e582b7708b2bee58432c793b9a7b100a9bc8/src/sys_fs_cgroup.c#L2115-L2148

Well, if only certain cores are assigned to the guest, then only these cores should have non-zero values. The way it is now, it wastes some memory in this case (for the unused cores), but the code is a lot simpler and faster. For the end user there should not be any difference. The dashboard hides by default all dimensions that are just zero.

**Comment by sn-x** [comment]
Hi! Thank you for the reply.

You are correct, that is the graph I was talking about and also the graph above it.
![cgroup cpu graphs](https://user-images.githubusercontent.com/15047082/27673707-6a6a5052-5ca2-11e7-8f0e-28732b839bb8.png)

A side question - is it currently possible to create cpu alerts on cgroups? If my guest has 4 cores (400%) and usage is above 320% (80%)? How would this be accomplished?

**Comment by ktsaou** [comment]
Something like this:

```
 alarm: high_cpu_on_my_vm
    on: cgroup_lxc_XXXXX.cpu
lookup: average -10s unaligned
 every: 10s
  warn: $this > 300
  crit: $this > 380
  info: my container uses too much cpu
    to: sysadmin
```


The `lookup` line calculates the last 10 seconds average of the sum of all dimensions of the chart and assigns the result to `$this`. The rest I guess are self explained.

**Comment by sn-x** [comment]
ktsao, you are doing the gods works. I am very impressed with netdata. I value your work and all the effort you have put in! One feature that I like very much are alerts with percentile thresholds. This pretty much eliminates user error (my error).

Consider this:
 - if we know that a namespace has 8G of memory available, we can reuse the percentile thold - 80%
 - if we know that a namespace has 4 cores available, we can reuse the percentile thold - 80%
Even if we increase/decrease allocated resources, tholds can stay the same, which would be a very big plus.

Like I said, I appreciate everything you have done! Maybe this can be treated as a proposal for improvement.

**Comment by ktsaou** [comment]
You are right. This is good.
Let's see how we can find this info...

**Comment by ktsaou** [comment]
@cakrit 	this is on our roadmap: extend the cgroups plugin to collect also the limits applied to cgroups, so that alarms about exhausted resources can be triggered.

**Comment by vlvkobal** [comment]
CPU and memory limits and alarms are implemented in #5172.

Disk limits can't be implemented for the existing charts. As far as cgroups define limits for every single disk but not for total bandwidth or IO operations for a cgroup, we need dynamic charts so that we would be able to define the corresponding alarms.

@ktsaou, what do you think? Should we implement dynamic charts for disks in cgroups? Is it a high priority?
